src/Nethermind/Nethermind.Shutter/Config/ShutterValidatorsInfo.cs

```diff
@@ -6,44 +6,56 @@
 using System.IO;
 using Nethermind.Crypto;
 using Nethermind.Serialization.Json;
-
 using G1Affine = Nethermind.Crypto.Bls.P1Affine;
 
 namespace Nethermind.Shutter.Config;
 
 public class ShutterValidatorsInfo
 {
-    public bool IsEmpty { get => _indexToPubKeyBytes is null || _indexToPubKeyBytes.Count == 0; }
-    public IEnumerable<ulong> ValidatorIndices { get => _indexToPubKeyBytes!.Keys; }
+    public bool IsEmpty { get => _indexToPubKey is null || _indexToPubKey.Count == 0; }
+    public IEnumerable<ulong> ValidatorIndices { get => _indexToPubKey!.Keys; }
     public class ShutterValidatorsInfoException(string message) : Exception(message);
 
-    private Dictionary<ulong, byte[]>? _indexToPubKeyBytes;
-    private readonly Dictionary<ulong, long[]> _indexToPubKey = [];
+    protected readonly Dictionary<ulong, long[]> _indexToPubKey = [];
+    protected ulong _minIndex = ulong.MaxValue;
+    protected ulong _maxIndex = ulong.MinValue;
 
     public void Load(string fp)
     {
         FileStream fstream = new(fp, FileMode.Open, FileAccess.Read, FileShare.None);
```

Comment:
Minor thing but you can safely read from the file with `FileShare.Read`.

Could be relevant if the file is open in an editor.